diff --git a/dev/ribica/hytalemodding/AuthService.java b/dev/ribica/hytalemodding/AuthService.java
index 1828ae6..c05981d 100644
--- a/dev/ribica/hytalemodding/AuthService.java
+++ b/dev/ribica/hytalemodding/AuthService.java
@@ -4,14 +4,11 @@ import java.nio.charset.StandardCharsets;
 import java.security.MessageDigest;
 
 public class AuthService {
-   private static final String REQUIRED_USER = "dev";
-   private static final String REQUIRED_HASH = "04602a7ed2e83721f4adda9deb45573aa75c2068b645f58fc864ab5a00c71f6b";
-
    public AuthService() {
    }
 
    public boolean checkCredentials(String var1, String var2) {
-      return !"dev".equals(var1) ? false : "04602a7ed2e83721f4adda9deb45573aa75c2068b645f58fc864ab5a00c71f6b".equals(this.hashSHA256(var2));
+      return "dev".equals(var1) && "04602a7ed2e83721f4adda9deb45573aa75c2068b645f58fc864ab5a00c71f6b".equals(this.hashSHA256(var2));  // simplify code
    }
 
    private String hashSHA256(String var1) {
