diff --git a/dev/ribica/hytalemodding/CustomTitleBar.java b/dev/ribica/hytalemodding/CustomTitleBar.java
index 020f710..447de00 100644
--- a/dev/ribica/hytalemodding/CustomTitleBar.java
+++ b/dev/ribica/hytalemodding/CustomTitleBar.java
@@ -27,27 +27,7 @@ public class CustomTitleBar extends JPanel {
       var4.setForeground(ThemeUtils.TEXT_GRAY);
       this.add(var4, "Center");
       if (var3) {
-         final JButton var5 = new JButton("X");
-         var5.setFocusPainted(false);
-         var5.setBorderPainted(false);
-         var5.setBackground(ThemeUtils.BG_HEADER);
-         var5.setForeground(ThemeUtils.TEXT_WHITE);
-         var5.setFont(new Font("Segoe UI", 1, 14));
-         var5.setPreferredSize(new Dimension(50, 35));
-         var5.addMouseListener(new MouseAdapter() {
-            {
-               Objects.requireNonNull(CustomTitleBar.this);
-            }
-
-            public void mouseEntered(MouseEvent var1) {
-               var5.setBackground(Color.decode("#d32f2f"));
-            }
-
-            public void mouseExited(MouseEvent var1) {
-               var5.setBackground(ThemeUtils.BG_HEADER);
-            }
-         });
-         var5.addActionListener((var1x) -> var1.dispose());
+         final JButton var5 = makeButton(var1);
          this.add(var5, "East");
       }
 
@@ -74,4 +54,29 @@ public class CustomTitleBar extends JPanel {
          }
       });
    }
+   // Extract to a method
+   private JButton makeButton(Window var1) {
+      final JButton var5 = new JButton("X");
+      var5.setFocusPainted(false);
+      var5.setBorderPainted(false);
+      var5.setBackground(ThemeUtils.BG_HEADER);
+      var5.setForeground(ThemeUtils.TEXT_WHITE);
+      var5.setFont(new Font("Segoe UI", 1, 14));
+      var5.setPreferredSize(new Dimension(50, 35));
+      var5.addMouseListener(new MouseAdapter() {
+         {
+            Objects.requireNonNull(CustomTitleBar.this);
+         }
+
+         public void mouseEntered(MouseEvent var1) {
+            var5.setBackground(Color.decode("#d32f2f"));
+         }
+
+         public void mouseExited(MouseEvent var1) {
+            var5.setBackground(ThemeUtils.BG_HEADER);
+         }
+      });
+      var5.addActionListener((var1x) -> var1.dispose());
+      return var5;
+   }
 }
