diff --git a/dev/ribica/hytalemodding/Launcher.java b/dev/ribica/hytalemodding/Launcher.java
index b629b83..a9023cf 100644
--- a/dev/ribica/hytalemodding/Launcher.java
+++ b/dev/ribica/hytalemodding/Launcher.java
@@ -55,7 +55,7 @@ public class Launcher extends JFrame {
 
    }
 
-   public static void main(String[] var0) {
+   public static void main(String[] args) {  // Decompile error
       ThemeUtils.applyDarkLook();
       SwingUtilities.invokeLater(() -> {
          Launcher var0 = new Launcher();
